modules/va_forms/app/sidekiq/va_forms/form_builder.rb

```diff
@@ -8,91 +8,124 @@ class FormBuilder
+    def check_form_validity(form, attrs, url)
+      response = fetch_form(url)
+      if response.success?
+        attrs[:valid_pdf] = true
+        attrs[:sha256] = Digest::SHA256.hexdigest(response.body)
+        attrs[:url] = url
 
-      form.update!(attrs)
-      form
+        send_slack_notifications(form, attrs, response.headers['Content-Type'])
+
+        attrs
+      else
+        raise FORM_FETCH_ERROR_MESSAGE
+      end
+    end
+
+    # Given a form +url+, makes a request for the form and returns the response
+    def fetch_form(url)
+      connection = Faraday.new(url) do |conn|
+        conn.use FaradayMiddleware::FollowRedirects
+        conn.options.open_timeout = 10
+        conn.options.timeout = 30
+        conn.adapter Faraday.default_adapter
+      end
+      connection.get
     end
 
     # Given a +existing_form+ instance of +VAForms::Form+, +updated_attributes+ for that form, and the +content_type+
-    # returned by the form URL, send appropriate Slack notifications
-    def send_notifications(existing_form, updated_attrs, content_type)
-      if updated_attrs[:valid_pdf]
-        if existing_form.url != updated_attrs[:url]
-          # If the URL has changed, we notify regardless of content
-          notify_slack(
-            "Form #{updated_attrs[:form_name]} has been updated.",
-            from_form_url: existing_form.url,
-            to_form_url: updated_attrs[:url]
-          )
-        elsif existing_form.sha256 != updated_attrs[:sha256] && content_type == 'application/pdf'
-          # If sha256 has changed, only notify if the URL actually returns a PDF, because if the URL is for a download
-          # web page instead, the change in sha256 may not actually reflect a change in the PDF itself
-          notify_slack("PDF contents of form #{updated_attrs[:form_name]} have been updated.")
-        end
-      elsif existing_form.valid_pdf
-        # If PDF is not valid, only notify if it was previously valid
+    # returned by the form URL, sends appropriate Slack notifications
+    def send_slack_notifications(existing_form, updated_attrs, content_type)
+      if existing_form.url != updated_attrs[:url]
+        # If the URL has changed, we notify regardless of content
         notify_slack(
-          "URL for form #{updated_attrs[:form_name]} no longer returns a valid PDF or web page.",
-          form_url: updated_attrs[:url]
+          "Form #{updated_attrs[:form_name]} has been updated.",
+          from_form_url: existing_form.url,
+          to_form_url: updated_attrs[:url]
         )
+      elsif existing_form.sha256 != updated_attrs[:sha256] && content_type == 'application/pdf'
+        # If sha256 has changed, only notify if the URL actually returns a PDF, because if the URL is for a download
+        # web page instead, the change in sha256 may not actually reflect a change in the PDF itself
+        notify_slack("PDF contents of form #{updated_attrs[:form_name]} have been updated.")
```

Comment:
This method was able to be simplified because the "no longer returns a valid PDF or web page" Slack message was moved to the `sidekiq_retries_exhausted` block.